instrumentation/activej-http-6.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/activejhttp/ActivejAsyncServletInstrumentation.java

```diff
@@ -52,39 +53,57 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class ServeAdvice {
 
-    @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void methodEnter(
-        @Advice.This AsyncServlet asyncServlet,
-        @Advice.Argument(0) HttpRequest request,
-        @Advice.Local("otelContext") Context context,
-        @Advice.Local("otelScope") Scope scope,
-        @Advice.Local("httpRequest") HttpRequest httpRequest) {
-      Context parentContext = currentContext();
-      httpRequest = request;
-      if (!instrumenter().shouldStart(parentContext, request)) {
-        return;
+    public static class AdviceScope {
+      private final HttpRequest httpRequest;
+      private final Context context;
+      private final Scope scope;
+
+      private AdviceScope(Context context, Scope scope, HttpRequest httpRequest) {
+        this.context = context;
+        this.scope = scope;
+        this.httpRequest = httpRequest;
+      }
+
+      @Nullable
+      public static AdviceScope start(HttpRequest request) {
+        Context parentContext = currentContext();
+        if (!instrumenter().shouldStart(parentContext, request)) {
+          return null;
+        }
+        Context context = instrumenter().start(parentContext, request);
+        return new AdviceScope(context, context.makeCurrent(), request);
+      }
+
+      public Promise<HttpResponse> end(Promise<HttpResponse> responsePromise, Throwable throwable) {
+        scope.close();
+        Promise<HttpResponse> returnValue = responsePromise;
+        if (throwable != null) {
+          instrumenter().end(context, httpRequest, null, throwable);
+        } else {
+          returnValue = PromiseWrapper.wrap(responsePromise, httpRequest, context);
+        }
+        return returnValue;
```

Comment:
Nit: unnecessary local variable.

```suggestion
        scope.close();
        if (throwable != null) {
          instrumenter().end(context, httpRequest, null, throwable);
          return responsePromise;
        } else {
          return PromiseWrapper.wrap(responsePromise, httpRequest, context);
        }
```